tests/acceptance/pageObjects/FilesPageElement/filesList.js

```diff
@@ -362,9 +373,35 @@ module.exports = {
       await this.api.elementIdLocation(firstRowElementId, (result) => {
         offset -= result.value.y
       })
-
       this.api.execute('scrollTo(0,' + offset + ')')
-      return this
+    },
+    checkFileName: async function (fileName, elementType = 'any') {
+      const fileSelector = this.getFileSelectorByFileName(fileName, elementType)
+      /**
+       *
+       * res.value return file/folder name and file extension in different row
+       * res.status return -1 if res return error otherwise return 1
+       */
+      let resourceName = null
+      let status = null
+      let error = null
+      await this.getText('xpath', fileSelector, (res) => {
+        status = res.status
+        const result = res.value
+        if (status === 0) {
+          resourceName = result.split('\n').join('')
+        } else if (status === -1) {
+          error = res.value.error
+        }
+      })
+      if (error !== null) {
+        assert.fail(error)
+      }
+      assert.strictEqual(
+        fileName,
+        resourceName,
+        `Expected file name to be "${fileName}" but found "${resourceName}"`
+      )
```

Comment:
it's not a good practice to perform assertions inside the `pageObjects` method.
you can return the text found from here and assert the expected case in the context files.

Reply:
waitForFileVisible() is calling checkFileName() to check file name and waitForFileVisible() is called by many other functions. So, we can't return the value from checkFileName()